src/pkg/caendr/caendr/models/sql/strain_annotated_variant.py

```diff
@@ -95,20 +93,21 @@ def column_default_visibility(col):
 
 
   @classmethod
-  def generate_interval_sql(cls, interval):
+  def generate_interval_sql(cls, interval, species=None):
     interval = interval.replace(',','')
     chrom = interval.split(':')[0]
     range = interval.split(':')[1]
     start = int(range.split('-')[0])
     stop = int(range.split('-')[1])
 
-    query = f"SELECT * FROM {cls.__tablename__} WHERE chrom='{chrom}' AND pos > {start} AND pos < {stop};"
+    species_clause = f"species_name='{species.name}' AND " if species else ''
+    query = f"SELECT * FROM {cls.__tablename__} WHERE {species_clause} chrom='{chrom}' AND pos > {start} AND pos < {stop};"
```

Comment:
this looks unsafe....can we build the query binding values instead of string concat?

Reply:
Fixed in https://github.com/AndersenLab/CAENDR/pull/373/commits/f69b630d07e43ff9898a96580f29df2a7ce5c2bd, with small updates in following 3 commits